source/runtime_gui.cpp

```diff
@@ -2444,21 +2446,65 @@ void reshade::runtime::draw_variable_editor()
 		// Draw preprocessor definition list after all uniforms of an effect file
 		std::string category_label = "Preprocessor definitions";
 		if (!_variable_editor_tabs)
-			for (float x = 0, space_x = ImGui::CalcTextSize(" ").x, width = (ImGui::CalcItemWidth() - ImGui::CalcTextSize(category_label.data()).x - 45) / 2; x < width; x += space_x)
+			for (float x = 0, space_x = ImGui::CalcTextSize(" ").x, width = (ImGui::CalcItemWidth() - ImGui::CalcTextSize(category_label.c_str()).x - 45) / 2; x < width; x += space_x)
 				category_label.insert(0, " ");
 
-		if (!_effects[effect_index].definitions.empty() &&
-			ImGui::TreeNodeEx(category_label.c_str(), ImGuiTreeNodeFlags_DefaultOpen | ImGuiTreeNodeFlags_NoTreePushOnOpen))
+		if (ImGui::TreeNodeEx(category_label.c_str(), ImGuiTreeNodeFlags_NoTreePushOnOpen))
 		{
-			for (const std::pair<std::string, std::string> &definition : _effects[effect_index].definitions)
+			if (!effect.preprocessed)
```

Comment:
I think this whole block of code can be replaced with this:
```c++
if (!effect.preprocessed)
    reload_effect = true;
```
This would avoid the code duplication here. The list will be empty for a single frame, but I think that is fine.